python/lsst/ts/externalscripts/auxtel/build_pointing_model.py

```diff
@@ -0,0 +1,290 @@
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+        self.log.debug(f"Configuration: {config}")
+
+        self.config = config
+
+        self.configure_grid()
+
+    def configure_grid(self):
+        """Configure the observation grid."""
+        self.log.debug("Configuring grid.")
+
+        npix = hp.nside2npix(self.config.nside)
+
+        healpy_indices = np.arange(npix)
+
+        azimuth, elevation = hp.pix2ang(
+            nside=self.config.nside, ipix=healpy_indices, lonlat=True
+        )
+
+        mask_out_of_range = np.bitwise_and(
+            elevation >= self.config.elevation_minimum,
+            elevation <= self.config.elevation_maximum,
+        )
+
+        self.elevation_grid = np.array(elevation[mask_out_of_range])
+        self.azimuth_grid = np.array(azimuth[mask_out_of_range])
+
+        self.log.debug("Sorting data in azimuth.")
+
+        azimuth_sort = np.argsort(self.azimuth_grid)
+
+        self.elevation_grid = self.elevation_grid[azimuth_sort]
+        self.azimuth_grid = self.azimuth_grid[azimuth_sort]
+
+        self.log.debug(f"Grid size: {self.grid_size}.")
+
+    @property
+    def grid_size(self):
+        return len(self.elevation_grid)
+
+    @property
+    def images_per_position(self):
+        return 2
+
+    @property
+    def camera_readout_time(self):
+        return 2.0
+
+    @property
+    def estimated_average_slew_time(self):
+        """The estimated average slew time considers a slew speed of 1 deg/sec.
+        With that in mind, it returs the resolution of the grid in degrees.
+        """
+        return hp.nside2resol(self.config.nside, arcmin=True) / 60.0
+
+    async def arun(self, checkpoint_active=False):
+
+        for grid_index, azimuth, elevation in zip(
+            range(self.grid_size), self.azimuth_grid, self.elevation_grid
+        ):
+
+            checkpoint_message = (
+                f"Execute grid position {grid_index+1} of {self.grid_size}: "
+                f"az={azimuth}, el={elevation}."
+            )
+
+            await self.handle_checkpoint(checkpoint_active, checkpoint_message)
+            await self.execute_grid(azimuth, elevation)
+
+    async def handle_checkpoint(self, checkpoint_active, checkpoint_message):
+
+        if checkpoint_active:
+            await self.checkpoint(checkpoint_message)
+        else:
+            self.log.info(checkpoint_message)
+
+    async def execute_grid(self, azimuth, elevation):
+        """Execute one grid position.
+
+        Parameters
+        ----------
+        azimuth : `float`
+            Azimuth of the grid position (in degrees).
+        elevation : `float`
+            Elevation of the grid position (in degrees).
+        """
+
+        try:
+            target = await self.atcs.find_target(
+                az=azimuth, el=elevation, mag_limit=self.config.magnitude_limit
+            )
+        except Exception:
```

Comment:
does it give any other information when it doesn't find a target? 
are there differnet exceptions that need to be handled differently? What it if can't connect or something?

Reply:
Yes, indeed it does issue different kinds of execeptions for different kinds of errors. I haven't figured out which is which tough.